src/app/services/mail/mail.service.ts

```diff
@@ -581,71 +596,154 @@ export class MailService {
 
   /**
    * Sends a email to the admin and collaborators of the form when the verified sms feature will be disabled.
-   * This happens only when the admin has hit a certain limit of sms verifications on his account
+   * This happens only when the admin has hit a certain limit of sms verifications on his account.
+   *
+   * Note that the email sent to the admin and collaborators will differ.
+   * This is because the admin will see all of their forms that are affected but collaborators
+   * only see forms which they are a part of.
+   *
    * @param form The form whose admin and collaborators will be issued the email
    * @returns ok(true) when mail sending is successful
    * @returns err(MailGenerationError) when there was an error in generating the html data for the mail
    * @returns err(MailSendError) when there was an error in sending the mail
    */
   sendSmsVerificationDisabledEmail = (
-    form: Pick<IPopulatedForm, 'permissionList' | 'admin' | 'title' | '_id'>,
+    form: Pick<IPopulatedForm, 'admin' | '_id'>,
   ): ResultAsync<true, MailGenerationError | MailSendError> => {
-    const htmlData: SmsVerificationDisabledData = {
-      formTitle: form.title,
-      formLink: `${this.#appUrl}/${form._id}`,
-      smsVerificationLimit: smsConfig.smsVerificationLimit,
+    // Step 1: Retrieve all public forms of admin that have sms verification enabled
+    return FormService.retrievePublicFormsWithSmsVerification(form.admin._id)
+      .andThen((forms) => {
+        // Step 2: Send the mail containing all the active forms to the admin
+        return this.sendDisabledMailForAdmin(forms, form.admin).map(() => forms)
+      })
+      .andThen((forms) => {
+        // Step 3: Send to each individual form
+        return combine(
+          forms.map((f) =>
+            // If there are no collaborators, do not send out the email.
+            // Admin would already have received a summary email from Step 2.
+            f.permissionList.length
+              ? this.sendDisabledMailForCollab(f, form.admin)
+              : okAsync<true, never>(true),
+          ),
+        )
+      })
+      .map(() => true)
+  }
+
+  // Helper method to send an email to all the collaborators of a given form that would be affected by
+  // Sms verifications being disabled for the form.
+  // Note that this method also emails the admin to notify them that the collaborators have been informed.
+  sendDisabledMailForCollab = (
+    form: IPopulatedForm,
+    admin: IPopulatedUser,
+  ): ResultAsync<true, MailGenerationError | MailSendError> => {
+    const htmlData: CollabSmsDisabledData = {
+      form: extractFormLinkView(form, this.#appUrl),
+      smsVerificationLimit:
+        smsConfig.smsVerificationLimit.toLocaleString('en-US'),
```

Comment:
If we are going to pass in "en-US" to the parameter, might as well just use `String(smsConfig.smsVerificationLimit)`?

Reply:
there's a (small) difference - using locale string formats big numbers for us, so numbers like 10000 turns into 10,000, which isn't present for using `String` (will give 10000 instead). 

I think i'd keep this cos admin will see this number and i want to make it look professional